### PR TITLE
Add OneHotEncoding Transform

### LINKED ISSUE #7: One hot encoding transform
A one-hot-encoding transform for categorical variables

MWE
```julia
x = ["foo", "bar", "baz"]

ohe = OneHotEncoding()

Transform.apply(x, ohe)

# output
3×3 Array{Int64,2}:
 1  0  0
 0  1  0
 0  0  1
```

### PR DESCRIPTION
**Pull request by nicoleepp**
Closes #7

### CODE REVIEW

File: src/one_hot_encoding.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Note that this Transform does not support specifying dims other than `:` (all dims) because
+it is a one-to-many transform (for example a `Vector` input produces a `Matrix` output).
+"""
+struct OneHotEncoding <: Transform
+    categories::Dict{Any, Int}
+
+    function OneHotEncoding(possible_values::AbstractVector)
+        if length(unique(possible_values)) < length(possible_values)
+            throw(ArgumentError("Expected an ordered list of all unique possible values"))
```
Comment (nicoleepp) [inline src/one_hot_encoding.jl:20]: Should test this

Comment (bencottier) [inline src/one_hot_encoding.jl:20]: Do the elements need to be a consistent order? Or does "ordered list" just refer to the type?

File: test/one_hot_encoding.jl [hunk 2]
```diff
@@ -0,0 +1,118 @@
+@testset "one hot encoding" begin
```
Comment (bencottier) [inline test/one_hot_encoding.jl:1]: As well as testing non-unique `categories` (as you noted), could we test
- data having a value not in `categories`
- case where `length(categories) > length(x)` - both (a) to test non-square matrix size both ways, and (b) to test `x` not having all the categories

**Review by bencottier — COMMENTED** [review]
I think we should have a couple more kinds of tests, it looks good otherwise.

File: src/one_hot_encoding.jl (hunk 1 above)
Comment (nicoleepp) [reply src/one_hot_encoding.jl:20]: They just need to be in the order of expected column output. so for input of 24 hours they should be ordered from Hour(1) to Hour(24). Then if it is applied on x = [Hour (2), Hour(2), Hour(1)] the result would be

```
 0  1  0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0  1  0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 1  0  0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 ```

**Review by bencottier — APPROVED** [review]

**Pull request merged**